### ISSUE TITLE
`rasa train core` invalidates nlu model

### ISSUE THREAD

**Issue by JEM-Mosig** [RasaHQ/rasa#7218]
**Rasa version**:

```
Rasa Version     : 2.0.3
Rasa SDK Version : 2.0.0
Rasa X Version   : 0.33.0
Python Version   : 3.7.0
Operating System : Linux-4.15.0-122-generic-x86_64-with-debian-buster-sid
Python Path      : /home/jem-mosig/miniconda3/envs/rasa-master/bin/python
```
From master, commit e2f6d028b64d8b1622bed81cd8b40e34cc27ced6.

**Issue**:

Training core by itself seems to invalidate the nlu model.

**Error (including full traceback)**:
```
/home/jem-mosig/rasa/rasa/rasa/shared/utils/io.py:93: UserWarning: Interpreter parsed an intent 'hi' which is not defined in the domain. Please make sure all intents are listed in the domain.
```

**Command or request that led to error**:
```
rasa init
<answer yes to train the model>
<answer yes to talk to the bot and see that it all works fine>
rasa train core
rasa shell
hi
```

**Comment by wochinge** [comment]
@JEM-Mosig Do I understand it right that you expected that you can use `rasa train nlu` first, then `rasa train core` and that this is the same as doing `rasa train`?

**Comment by JEM-Mosig** [comment]
@wochinge I expected the bot to work if I do `rasa train` and then `rasa train core`. But unrelated to that, I'd also expect it to work with `rasa train nlu` and then `rasa train core`. Is that a wrong assumption?

**Comment by wochinge** [comment]
Yes, I'd actually would rather see this as a new feature than a bug. What is your use case to train them independently? `rasa train` will automatically skip training NLU if there were no changes (there are currently some hiccups with that 😬 )

**Comment by JEM-Mosig** [comment]
Ah, ok. Then I'll just use `rasa train`. I think I just ran into those "hiccups" there when building my bots, so it keeps re-training NLU even though I didn't change anything on that side, which is why I tried `rasa train core`.

**Comment by wochinge** [comment]
I think that's related due to the issue which Tom brought up. The hashes we calculate aren't stable across multiple runs. We will revisit this logic for end-to-end as well 👍 Closing this one then.

**Comment by wochinge** [comment]
Btw, the issues with the hashs should be fixed in Rasa Open Source 2.1: https://github.com/RasaHQ/rasa/pull/7246/files
